UPDATE: Create and Publish Custom GitHub Actions

a-a-ron opened the pull request:
The purpose of this PR is to see how we can increase the completion rate and improve the user experience. Below is the August data for this module:

## Module stats

| Module | Completion Rate | # completed / # started |
|--------|--------|--------|
| [Create and publish custom GitHub Actions](https://learn.microsoft.com/en-us/training/modules/create-custom-github-actions/?ns-enrollment-type=Collection&ns-enrollment-id=n5p4a5z7keznp5) | 15.25% | 12/116 |

**For reference:** See the [parent issue](https://github.com/githubpartners/microsoft-learn/issues/35) for context

## Module file structure
<img width="296" alt="Screen Shot 2022-09-28 at 11 45 09 AM" src="https://user-images.githubusercontent.com/6351798/192852614-dfb7a1d2-14d4-4331-8db1-f61c85cf8649.png">

## Items to review for this module

We need to conduct a performance review on each of these modules to see how we can increase the completion rates, improve the user experience, and maintain the level of quality that we set for our learners. 

### Items for consideration
- **Thoroughly review the module content for flow, narrative, and length**
  - Review use of images, tables, and other visual elements
  - Does the content read like a doc or like a conversation guide?
  - Does the estimated time to complete each unit remain accurate?
- **Review the introduction unit** 
  - Does it capture your attention?
  - Does it contain repetitive content from the course title page?
  - Do the learning objectives/scope align with the content? 
  - Does the target audience match the content?
  - Do we accurately describe the module and learning objectives? (Is it possible that learners start the course thinking it's one thing to find out it's another?) 
- **Review the module exercise**
  - Does it work?
  - Does it enforce a learning objective?
  - How is it placed in the module? Should we move it around?
  - Do we need other exercises?
- **Review the questions at the end of the module**
  - Do the questions test on content taught in the module?
  - Do the questions follow best practices for item writing?
 
## Estimated Work Effort
| Task | Hours | 
|--------|--------|
| Detailed module review | 18 hours | 
| Action items as identified from review | TBD | 

## Action items for this module

After a detailed review of the module, below are the items we need to add/remove/modify to improve the learner experience for this module.

🚧 {**Plan of action [in comment below](https://github.com/githubpartners/microsoft-learn/pull/45#issuecomment-1261281453)**}🚧

a-a-ron committed:
Update index.yml

rmallorybpc commented:
| Module | Completion Rate | # completed / # started |
|--------|--------|--------|
| [Create and publish custom GitHub Actions](https://learn.microsoft.com/en-us/training/modules/create-custom-github-actions/?ns-enrollment-type=Collection&ns-enrollment-id=n5p4a5z7keznp5) | 15.25% | 12/116 |

## Plan of action
- **Review the module launch page unit**
     - Does it capture your attention?
       - The introduction paragraph is very detailed and it has long sentences with a lot of learning objectives. The introduction paragraph and learning objectives completely overlap. Decrease the opening paragraph content.
       - The learning objectives can be reworked to better align with the topics in the units and engage the user more.
- **Review the introduction unit**
  - Does it capture your attention?
    - The introduction is too long and does not create a story for the learner.
  - Do the learning objectives/scope align with the content? 
    - The introduction does not lead into the learning objectives, and then connect to the unit titles.
  - Does the target audience match the content?
    - Is there a chance that a learning who knows workflows and CI would know this information?
  - Do we accurately describe the module and learning objectives? (Is it possible that learners start the course thinking it's one thing to find out it's another?)
    - Better story-like opening sentence.
- **Thoroughly review the module content for flow, narrative, and length**
  - Review use of images, tables, and other visual elements
    - The images are good.
  - Does the content read like a doc or like a conversation guide?
    - Unit 3 Publish a custom GitHub action looks good.
    - The length matches other modules.
- **Review the module exercise**
  - The exercise on the launch page says that the exercise takes 6 min. The exercise states, "How long: This course is 6 steps long and takes about 1 to 2 hours to be completed." This could be a big stopping point.
  - Does it work?
  - Does it enforce a learning objective?
    - The exercise is writing a GitHub JavaScript Action and automation. The learning should point out the Javascript a bit more.
  - How is it placed in the module? Should we move it around?
  - Do we need other exercises?
- **Review the questions at the end of the module**
  - Do the questions test on content taught in the module?
    - Q1 - Ruby is not mentioned in the units.
    - Q2 - is fine.
    - Q3 - Options 1+4 are super close. I would write 4 as, echo "::errorThis is an error message::". That would be a bit more clear as incorrect.
  - Do the questions follow best practices for item writing?
    - The writing overall is ok, with the notes above.

rmallorybpc commented:
# Move Introduction information

## Learning objectives

In this module, you will:

- Identify the metadata and syntax needed to create custom GitHub actions
- Learn best practices to publish a private or public GitHub action
- Consume actions within a workflow file
- Create custom JavaScript based actions
- Publish a newly created action to the GitHub Marketplace

## Prerequisites

This module assumes you have basic familiarity with GitHub Actions and workflows.

- A GitHub account
- The ability to navigate and edit files in GitHub
  - For more information about GitHub, see [Introduction to GitHub](https://lab.github.com/githubtraining/introduction-to-github?azure-portal=true).
- Basic familiarity with GitHub Actions and workflows
  - If you aren't familiar with workflows, jobs and steps, check out the [Automate development tasks by using GitHub Actions](/learn/modules/github-actions-automate-tasks/) module.
- Basic familiarity with continuous integration using GitHub Actions and workflows
  - If you're unfamiliar with continuous integration using GitHub Actions and workflows, check out [Build continuous integration workflows by using GitHub Actions](/learn/modules/github-actions-ci/)

rmallorybpc committed:
Update introduction.md

Remove duplicate information from the Introduction unit.

rmallorybpc committed:
Update index.yml

Move Learning Objectives and Prerequisite from the Introduction to the Launch page (index.yml file).

rmallorybpc committed:
Update introduction.md

Update the Introduction to start the learning story.

rmallorybpc committed:
Update introduction.md

Quick rework edit.

rmallorybpc committed:
Update introduction.md

Minor editing.

rmallorybpc committed:
Update index.yml

Reworked the learning objectives.

rmallorybpc committed:
Update create-custom-github-action.md

Edit to the order of the opening paragraph.

rmallorybpc committed:
Update introduction.md

Edits for learner flow.

rmallorybpc committed:
Update create-custom-github-action.md

End flow edits.

rmallorybpc committed:
Update publish-custom-github-action.md

End flow edits.

rmallorybpc committed:
Update exercise-create-custom-action.yml

Updated the length information.

rmallorybpc committed:
Update knowledge-check.yml

Updated question one, anwers, and explanations because Ruby was not discussed in the unit contents. Updated answer 4 in question 3 because it was too close to answer 1.

rmallorybpc committed:
Update introduction.md

Added the prerequisites and learning objectives back
